### ISSUE TITLE
Regression loss and metric functions to module

### ISSUE THREAD

**Issue by FrancescaDr** [theislab/interscale#34]
### Description of feature

Add loss and metric function calculations to module. They can be called in `forward(compute_loss: bool)`, True if training and False if testing. 

For regression the loss and metric can be calculated either across cells (for each gene) or across genes (for each cell). In [Schwessinger et al., 2023](https://doi.org/10.1101/2023.07.26.550634) describe the advantages for cross-cell and cross-gene and define a [BalancedPearsonCorr](https://github.com/GSK-AI/seq2cells/blob/main/seq2cells/metrics_and_losses/losses.py) loss that considers both. 

Try the different loss functions to see which one is best for our model learning:
- [x] Cross cell: Most common
- [x] Cross gene
- [x] Balanced (both cross cell and cross gene)

Write test function for Pearson corelation. Compare whether using BalancedPearsonCorr works if I put relative weight to 1.0/0.0 respectively. 

### Previous code 

Put previous code in functions to test the outcome. 

```Python
if self.cross_corr == 'gene':
            # score per cell, cell numbers dependent on sliding windows / spatial slide
            #metrics = self._setup_regression_metrics(nr_cells) # does not take nr_cells as input anymore   
            if self.loss_type == 'GaussianNLL':
                loss = self.loss(y_pred, y_true, y_var)
            else:
                loss = self.loss(y_pred, y_true)
            y_pred = y_pred.T.contiguous()
            y_true = y_true.T.contiguous()
            assert y_pred.shape[0] == self.module.n_input
        elif self.cross_corr == 'cell':
            if self.loss_type == 'GaussianNLL':
                loss = self.loss(y_pred.T.contiguous(), y_true.T.contiguous(), y_var) # loss calculated over [N,:]
            else:
                loss = self.loss(y_pred.T.contiguous(), y_true.T.contiguous()) # loss calculated over [N,:]
            assert y_pred.shape[1] == self.module.n_input
            
        metrics = metrics(y_pred, y_true)
```

**Comment by FrancescaDr** [comment]
Multi-modal loss function not possible
